src/oc/storage/resources/entry.clj

```diff
@@ -308,6 +318,22 @@
   (delete-version conn uuid)
   (db-common/delete-resource conn table-name uuid))
 
+(schema/defn ^:always-validate revert-entry!
+  "Given the UUID of the entry and revision, replace current revision with specified. Return `true` on success."
+  [conn entry entry-version user]
+  {:pre [(db-common/conn? conn)]}
+  (let [new-entry (dissoc entry-version
+                          :revision-id
+                          :version-uuid
+                          :revision-date
+                          :revision-author)]
+    ;; if version number is 0 delete the actual entry
+    (if (zero? (:revision-id entry-version))
+      (do
+        (delete-entry! conn (:uuid entry-version))
```

Comment:
Maybe i understand why on the resource `exists?` we check if revision-id is zero and if so we return the current version.
It doesn't matter what version we return there since the entry will be deleted later.
That's right?

Reply:
right :)